### PR TITLE
Fix undefined workspace variable error if no enclosing workspace has been found

### PR DESCRIPTION
**Pull request by meyerj**
Fixes a Python error introduced in https://github.com/Intermodalics/catkin_tools/pull/3: Without a `workspace` argument (`-w`) and if no workspace was found by `find_enclosing_workspaces(getcwd())`, the `workspace` variable was undefined in [line 186](https://github.com/Intermodalics/catkin_tools/blob/b6867b190d40fc1d1e024e014523bf6884967af4/catkin_tools/context.py#L186).

### CODE REVIEW

File: catkin_tools/metadata.py [hunk 1]
```diff
@@ -232,16 +232,6 @@ def init_metadata_root(workspace_path, reset=False):
...
 
-    # Check if the desired workspace is enclosed in another workspace
-    marked_workspace = find_enclosing_workspaces(workspace_path)
-    if marked_workspace: marked_workspace = marked_workspace[0]
-
-    if marked_workspace and marked_workspace != workspace_path:
-        raise IOError(
-            "Can't initialize Catkin workspace in path %s because it is "
-            "already contained in another workspace: %s." %
-            (workspace_path, marked_workspace))
```
Comment (PerrineAguiar) [inline catkin_tools/metadata.py:243]: Why removing this part? It does not seem related to the same issue.

**Review by PerrineAguiar — APPROVED** [review]

File: catkin_tools/metadata.py (hunk 1 above)
Comment (meyerj) [reply catkin_tools/metadata.py:243]: That code part is triggered by `catkin init` and it checks whether a workspace is enclosed in another workspace. If yes, `catkin` refuses to initialize a new workspace, even if you ask for it explicitly with `catkin init -w path/to/sub/workspace`.

In our patched version (with https://github.com/Intermodalics/catkin_tools/pull/3) that is fine, and I sometimes use this for tests in a temporary nested workspace, for example for a single package or for a submodule. But by default `catkin` will always find and use the enclosing workspace, so you have to pass `-w` explicitly when referring to the nested workspace.

But we could print a warning only, instead of removing the check completely...
